src/Microsoft.ML.LightGBM/WrappedLightGbmBooster.cs

```diff
@@ -96,10 +96,10 @@ internal unsafe string GetModelString()
 
         private static double[] Str2DoubleArray(string str, char delimiter)
         {
-            return str.Split(delimiter).Select(
-                x => { double t; double.TryParse(x, out t); return t; }
-            ).ToArray();
-
+            return str.Split(delimiter)
+                      .Select(s => double.TryParse(s.Replace("inf", "∞"), out double rslt) ? rslt :
+                                    (s.Contains("nan") ? double.NaN : throw new Exception($"Cannot parse as double: {s}")))
```

Comment:
>throw new Exception($"Cannot parse as double: {s}")) [](start = 70, length = 52)

please, never throw unmarked exception in your code.

Reply:
I would probably either skip it completely or just return default value for double as we had before.

---
In reply to: [243433489](https://github.com/dotnet/machinelearning/pull/1934#discussion_r243433489) [](ancestors = 243433489)

Reply:
Question: do we want to fully mimic LightGBM behavior? If yes, we need this change. No, we can just close this PR.

---
In reply to: [243433959](https://github.com/dotnet/machinelearning/pull/1934#discussion_r243433959) [](ancestors = 243433959,243433489)

Reply:
I'm not against your PR, I'm against throwing unmarked exception.
Sorry for my inability to express my thoughts in proper understandable way.
>I would probably either skip it completely or just return default value for double as we had before

I don't mean your whole new code, I mean case where we can't parse double.
We have two options for that case:
- Throw MARKED exception like `throw env.Exception(message)`
- Use default value for double.

Reply:
If it is expected that this throws, you should probably explicitly check the strings for `inf/-inf/nan` (https://github.com/dotnet/machinelearning/pull/1934#discussion_r243441276) and fallback to just calling `double.Parse` (which will throw the appropriate format exception)